# Import libraries

In [1]:
import pandas as pd
import numpy as np

In [2]:
labeled = pd.read_csv('data/labeled_data.csv')

labeled_cn7 = pd.read_csv('data/moldset_labeled_cn7.csv')
labeled_rg3 = pd.read_csv('data/moldset_labeled_rg3.csv')
moldset_labeled = pd.read_csv('data/moldset_labeled.csv')

unlabeled_cn7 = pd.read_csv('data/moldset_unlabeled_cn7.csv')
unlabeled_rg3 = pd.read_csv('data/moldset_unlabeled_rg3.csv')

supervised_cn7 = pd.read_csv('data/supervised_label_cn7.csv')

unlabeled = pd.read_csv('data/unlabeled_data.csv')

In [3]:
pd.set_option('display.max_columns', None)
pd.set_option('display.max_colwidth', None)
# pd.set_option('display.max_rows', None)

# Load data

In [4]:
# shape, info, head를 출력하는 함수

def inspect_df(df, name):
    print("=" * 80)
    print(f"📌 {name}")
    print("=" * 80)

    # Shape
    print(f"\n[Shape]")
    print(df.shape)

    # Info
    print(f"\n[Info]")
    df.info()

    # Head
    display(df.head())

    print("\n")

### `labeled_data.csv`

In [5]:
inspect_df(labeled, "Labeled Data")

📌 Labeled Data

[Shape]
(7996, 45)

[Info]
<class 'pandas.DataFrame'>
RangeIndex: 7996 entries, 0 to 7995
Data columns (total 45 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   _id                       7996 non-null   str    
 1   TimeStamp                 7996 non-null   str    
 2   PART_FACT_PLAN_DATE       7996 non-null   str    
 3   PART_FACT_SERIAL          7996 non-null   int64  
 4   PART_NAME                 7996 non-null   str    
 5   EQUIP_CD                  7996 non-null   str    
 6   EQUIP_NAME                7996 non-null   str    
 7   PassOrFail                7996 non-null   str    
 8   Reason                    71 non-null     str    
 9   Injection_Time            7996 non-null   float64
 10  Filling_Time              7996 non-null   float64
 11  Plasticizing_Time         7996 non-null   float64
 12  Cycle_Time                7996 non-null   float64
 13  Clamp_Close_Time          7996 

,_id,TimeStamp,PART_FACT_PLAN_DATE,PART_FACT_SERIAL,PART_NAME,EQUIP_CD,EQUIP_NAME,PassOrFail,Reason,Injection_Time,Filling_Time,Plasticizing_Time,Cycle_Time,Clamp_Close_Time,Cushion_Position,Switch_Over_Position,Plasticizing_Position,Clamp_Open_Position,Max_Injection_Speed,Max_Screw_RPM,Average_Screw_RPM,Max_Injection_Pressure,Max_Switch_Over_Pressure,Max_Back_Pressure,Average_Back_Pressure,Barrel_Temperature_1,Barrel_Temperature_2,Barrel_Temperature_3,Barrel_Temperature_4,Barrel_Temperature_5,Barrel_Temperature_6,Barrel_Temperature_7,Hopper_Temperature,Mold_Temperature_1,Mold_Temperature_2,Mold_Temperature_3,Mold_Temperature_4,Mold_Temperature_5,Mold_Temperature_6,Mold_Temperature_7,Mold_Temperature_8,Mold_Temperature_9,Mold_Temperature_10,Mold_Temperature_11,Mold_Temperature_12
0,5f8928bb9c0189cc666ef19b,2020-10-16 04:57:47,2020-10-16 오전 12:00:00,24,CN7 W/S SIDE MLD'G RH,S14,650톤-우진2호기,Y,NaN,9.59,4.47,16.92,59.520000,7.13,653.409973,0.0,68.849998,647.98999,55.400002,30.700001,292.5,141.800003,136.800003,37.400002,59.299999,276.500000,274.700012,274.799988,269.200012,255.000000,229.699997,0.0,66.300003,0.0,0.0,24.799999,27.5,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
1,5f8928de9c0189cc666ef20b,2020-10-16 04:58:48,2020-10-16 오전 12:00:00,24,CN7 W/S SIDE MLD'G RH,S14,650톤-우진2호기,Y,NaN,9.60,4.48,16.91,59.580002,7.13,653.409973,0.0,68.839996,647.98999,55.299999,30.799999,292.5,141.800003,136.899994,37.500000,59.299999,276.200012,275.500000,275.299988,270.799988,254.699997,229.500000,0.0,67.199997,0.0,0.0,24.799999,27.6,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
2,5f8928df9c0189cc666ef213,2020-10-16 04:58:48,2020-10-16 오전 12:00:00,23,CN7 W/S SIDE MLD'G LH,S14,650톤-우진2호기,Y,NaN,9.60,4.48,16.91,59.580002,7.13,653.409973,0.0,68.839996,647.98999,55.299999,30.799999,292.5,141.800003,136.899994,37.500000,59.299999,276.200012,275.500000,275.299988,270.799988,254.699997,229.500000,0.0,67.199997,0.0,0.0,24.799999,27.6,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
3,5f8928f39c0189cc666ef25e,2020-10-16 04:59:48,2020-10-16 오전 12:00:00,23,CN7 W/S SIDE MLD'G LH,S14,650톤-우진2호기,Y,NaN,9.59,4.48,16.91,59.560001,7.13,653.419983,0.0,68.839996,647.98999,55.299999,31.000000,292.5,141.800003,136.800003,37.500000,59.299999,276.500000,275.000000,275.399994,271.100006,254.899994,230.000000,0.0,66.900002,0.0,0.0,25.000000,27.6,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
4,5f8928f59c0189cc666ef265,2020-10-16 04:59:48,2020-10-16 오전 12:00:00,24,CN7 W/S SIDE MLD'G RH,S14,650톤-우진2호기,Y,NaN,9.59,4.48,16.91,59.560001,7.13,653.419983,0.0,68.839996,647.98999,55.299999,31.000000,292.5,141.800003,136.800003,37.500000,59.299999,276.500000,275.000000,275.399994,271.100006,254.899994,230.000000,0.0,66.900002,0.0,0.0,25.000000,27.6,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0


### `moldset_labeled_cn7.csv`, `moldset_labeled_rg3.csv`, `moldset_labeled.csv`

In [6]:
inspect_df(labeled_cn7, "Moldset Labeled CN7 Data")

📌 Moldset Labeled CN7 Data

[Shape]
(1211, 26)

[Info]
<class 'pandas.DataFrame'>
RangeIndex: 1211 entries, 0 to 1210
Data columns (total 26 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Unnamed: 0                1211 non-null   int64  
 1   PassOrFail                1211 non-null   int64  
 2   Injection_Time            1211 non-null   float64
 3   Filling_Time              1211 non-null   float64
 4   Plasticizing_Time         1211 non-null   float64
 5   Cycle_Time                1211 non-null   float64
 6   Clamp_Close_Time          1211 non-null   float64
 7   Cushion_Position          1211 non-null   float64
 8   Plasticizing_Position     1211 non-null   float64
 9   Clamp_Open_Position       1211 non-null   float64
 10  Max_Injection_Speed       1211 non-null   float64
 11  Max_Screw_RPM             1211 non-null   float64
 12  Average_Screw_RPM         1211 non-null   float64
 13  Max_Injection_Press

,Unnamed: 0,PassOrFail,Injection_Time,Filling_Time,Plasticizing_Time,Cycle_Time,Clamp_Close_Time,Cushion_Position,Plasticizing_Position,Clamp_Open_Position,Max_Injection_Speed,Max_Screw_RPM,Average_Screw_RPM,Max_Injection_Pressure,Max_Switch_Over_Pressure,Max_Back_Pressure,Average_Back_Pressure,Barrel_Temperature_1,Barrel_Temperature_2,Barrel_Temperature_3,Barrel_Temperature_4,Barrel_Temperature_5,Barrel_Temperature_6,Hopper_Temperature,Mold_Temperature_3,Mold_Temperature_4
0,0,0,1.679123,1.491397,-0.478017,0.361047,0.97858,-0.711164,1.532018,0.0,-1.152398,-0.795730,-0.132022,0.63521,1.256518,-0.916676,-0.533773,0.729073,-1.598068,-0.672350,-1.536620,0.008133,-1.352970,-1.780463,1.023094,1.287574
1,1,0,2.080358,1.894372,-0.527100,1.886261,0.97858,-0.711164,1.471608,0.0,-1.347095,0.049553,-0.132022,0.63521,1.573231,-0.740752,-0.533773,0.056153,0.622557,0.914107,0.504051,-0.915098,-2.263700,-0.874018,1.023094,1.337339
2,2,0,2.080358,1.894372,-0.527100,1.886261,0.97858,-0.711164,1.471608,0.0,-1.347095,0.049553,-0.132022,0.63521,1.573231,-0.740752,-0.533773,0.056153,0.622557,0.914107,0.504051,-0.915098,-2.263700,-0.874018,1.023094,1.337339
3,3,0,1.679123,1.894372,-0.527100,1.377857,0.97858,1.202502,1.471608,0.0,-1.347095,1.740153,-0.132022,0.63521,1.256518,-0.740752,-0.533773,0.729073,-0.765355,1.231418,0.886706,-0.299626,0.013158,-1.176164,1.166162,1.337339
4,4,0,1.679123,1.894372,-0.527100,1.377857,0.97858,1.202502,1.471608,0.0,-1.347095,1.740153,-0.132022,0.63521,1.256518,-0.740752,-0.533773,0.729073,-0.765355,1.231418,0.886706,-0.299626,0.013158,-1.176164,1.166162,1.337339


In [7]:
inspect_df(labeled_rg3, "Moldset Labeled RG3 Data")

📌 Moldset Labeled RG3 Data

[Shape]
(1182, 26)

[Info]
<class 'pandas.DataFrame'>
RangeIndex: 1182 entries, 0 to 1181
Data columns (total 26 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Unnamed: 0                1182 non-null   int64  
 1   PassOrFail                1182 non-null   int64  
 2   Injection_Time            1182 non-null   float64
 3   Filling_Time              1182 non-null   float64
 4   Plasticizing_Time         1182 non-null   float64
 5   Cycle_Time                1182 non-null   float64
 6   Clamp_Close_Time          1182 non-null   float64
 7   Cushion_Position          1182 non-null   float64
 8   Plasticizing_Position     1182 non-null   float64
 9   Clamp_Open_Position       1182 non-null   float64
 10  Max_Injection_Speed       1182 non-null   float64
 11  Max_Screw_RPM             1182 non-null   float64
 12  Average_Screw_RPM         1182 non-null   float64
 13  Max_Injection_Press

,Unnamed: 0,PassOrFail,Injection_Time,Filling_Time,Plasticizing_Time,Cycle_Time,Clamp_Close_Time,Cushion_Position,Plasticizing_Position,Clamp_Open_Position,Max_Injection_Speed,Max_Screw_RPM,Average_Screw_RPM,Max_Injection_Pressure,Max_Switch_Over_Pressure,Max_Back_Pressure,Average_Back_Pressure,Barrel_Temperature_1,Barrel_Temperature_2,Barrel_Temperature_3,Barrel_Temperature_4,Barrel_Temperature_5,Barrel_Temperature_6,Hopper_Temperature,Mold_Temperature_3,Mold_Temperature_4
0,1211,0,-0.029099,-1.192531,1.977737,0.249511,1.033078,2.867608,-1.755816,0.0,2.187583,0.677109,-0.421798,-2.077961,-1.184902,-1.330250,-1.306878,0.096629,-0.496175,-0.002081,0.047749,0.282727,-0.354184,0.183232,-1.393485,-1.576042
1,1212,0,-0.029099,-1.192531,1.977737,0.249511,1.033078,2.867608,-1.755816,0.0,2.187583,0.677109,-0.421798,-2.077961,-1.184902,-1.330250,-1.306878,0.096629,-0.496175,-0.002081,0.047749,0.282727,-0.354184,0.183232,-1.393485,-1.576042
2,1213,0,-0.029099,-1.192531,1.528778,0.249511,1.033078,2.075089,-0.192928,0.0,2.187583,0.677109,-0.421798,-1.421642,-0.974786,-1.019133,-0.841877,0.562844,1.442567,-0.002081,0.340262,1.850096,-0.851297,0.183232,-1.535968,-1.576042
3,1214,0,-0.029099,-1.192531,1.528778,0.249511,1.033078,2.075089,-0.192928,0.0,2.187583,0.677109,-0.421798,-1.421642,-0.974786,-1.019133,-0.841877,0.562844,1.442567,-0.002081,0.340262,1.850096,-0.851297,0.183232,-1.535968,-1.576042
4,1215,0,-0.029099,-1.192531,0.630880,-0.298708,1.033078,1.277707,-1.755816,0.0,1.402554,0.677109,2.655764,-0.765222,-0.554587,-0.708027,-1.539374,1.262095,-0.253813,-0.002081,0.925242,0.282727,-1.348485,0.284910,-1.535968,-1.576042


In [8]:
inspect_df(moldset_labeled, "Moldsest Labeled Data")

📌 Moldsest Labeled Data

[Shape]
(2607, 47)

[Info]
<class 'pandas.DataFrame'>
RangeIndex: 2607 entries, 0 to 2606
Data columns (total 47 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Unnamed: 0                2607 non-null   int64  
 1   _id                       2607 non-null   str    
 2   TimeStamp                 2607 non-null   str    
 3   PART_FACT_PLAN_DATE       2607 non-null   str    
 4   PART_FACT_SERIAL          2607 non-null   int64  
 5   PART_NO                   2607 non-null   str    
 6   PART_NAME                 2607 non-null   str    
 7   EQUIP_CD                  2607 non-null   str    
 8   EQUIP_NAME                2607 non-null   str    
 9   PassOrFail                2607 non-null   int64  
 10  Reason                    52 non-null     str    
 11  Injection_Time            2607 non-null   float64
 12  Filling_Time              2607 non-null   float64
 13  Plasticizing_Time     

,Unnamed: 0,_id,TimeStamp,PART_FACT_PLAN_DATE,PART_FACT_SERIAL,PART_NO,PART_NAME,EQUIP_CD,EQUIP_NAME,PassOrFail,Reason,Injection_Time,Filling_Time,Plasticizing_Time,Cycle_Time,Clamp_Close_Time,Cushion_Position,Switch_Over_Position,Plasticizing_Position,Clamp_Open_Position,Max_Injection_Speed,Max_Screw_RPM,Average_Screw_RPM,Max_Injection_Pressure,Max_Switch_Over_Pressure,Max_Back_Pressure,Average_Back_Pressure,Barrel_Temperature_1,Barrel_Temperature_2,Barrel_Temperature_3,Barrel_Temperature_4,Barrel_Temperature_5,Barrel_Temperature_6,Barrel_Temperature_7,Hopper_Temperature,Mold_Temperature_1,Mold_Temperature_2,Mold_Temperature_3,Mold_Temperature_4,Mold_Temperature_5,Mold_Temperature_6,Mold_Temperature_7,Mold_Temperature_8,Mold_Temperature_9,Mold_Temperature_10,Mold_Temperature_11,Mold_Temperature_12
0,0,5f8928bb9c0189cc666ef19b,2020-10-16 04:57:47,2020-10-16 오전 12:00:00,24,86141AA000,CN7 W/S SIDE MLD'G RH,S14,650톤-우진2호기,0,NaN,9.59,4.47,16.92,59.520000,7.13,653.409973,0.0,68.849998,647.98999,55.400002,30.700001,292.5,141.800003,136.800003,37.400002,59.299999,276.500000,274.700012,274.799988,269.200012,255.000000,229.699997,0.0,66.300003,0.0,0.0,24.799999,27.5,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
1,1,5f8928de9c0189cc666ef20b,2020-10-16 04:58:48,2020-10-16 오전 12:00:00,24,86141AA000,CN7 W/S SIDE MLD'G RH,S14,650톤-우진2호기,0,NaN,9.60,4.48,16.91,59.580002,7.13,653.409973,0.0,68.839996,647.98999,55.299999,30.799999,292.5,141.800003,136.899994,37.500000,59.299999,276.200012,275.500000,275.299988,270.799988,254.699997,229.500000,0.0,67.199997,0.0,0.0,24.799999,27.6,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
2,2,5f8928df9c0189cc666ef213,2020-10-16 04:58:48,2020-10-16 오전 12:00:00,23,86131AA000,CN7 W/S SIDE MLD'G LH,S14,650톤-우진2호기,0,NaN,9.60,4.48,16.91,59.580002,7.13,653.409973,0.0,68.839996,647.98999,55.299999,30.799999,292.5,141.800003,136.899994,37.500000,59.299999,276.200012,275.500000,275.299988,270.799988,254.699997,229.500000,0.0,67.199997,0.0,0.0,24.799999,27.6,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
3,3,5f8928f39c0189cc666ef25e,2020-10-16 04:59:48,2020-10-16 오전 12:00:00,23,86131AA000,CN7 W/S SIDE MLD'G LH,S14,650톤-우진2호기,0,NaN,9.59,4.48,16.91,59.560001,7.13,653.419983,0.0,68.839996,647.98999,55.299999,31.000000,292.5,141.800003,136.800003,37.500000,59.299999,276.500000,275.000000,275.399994,271.100006,254.899994,230.000000,0.0,66.900002,0.0,0.0,25.000000,27.6,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
4,4,5f8928f59c0189cc666ef265,2020-10-16 04:59:48,2020-10-16 오전 12:00:00,24,86141AA000,CN7 W/S SIDE MLD'G RH,S14,650톤-우진2호기,0,NaN,9.59,4.48,16.91,59.560001,7.13,653.419983,0.0,68.839996,647.98999,55.299999,31.000000,292.5,141.800003,136.800003,37.500000,59.299999,276.500000,275.000000,275.399994,271.100006,254.899994,230.000000,0.0,66.900002,0.0,0.0,25.000000,27.6,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0


### `moldset_unlabeled_cn7.csv`, `moldset_unlabeled_rg3.csv`

In [9]:
inspect_df(unlabeled_cn7, "Moldset Unlabeled CN7 Data")

📌 Moldset Unlabeled CN7 Data

[Shape]
(35239, 25)

[Info]
<class 'pandas.DataFrame'>
RangeIndex: 35239 entries, 0 to 35238
Data columns (total 25 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Unnamed: 0                35239 non-null  int64  
 1   Injection_Time            35239 non-null  float64
 2   Filling_Time              35239 non-null  float64
 3   Plasticizing_Time         35239 non-null  float64
 4   Cycle_Time                35239 non-null  float64
 5   Clamp_Close_Time          35239 non-null  float64
 6   Cushion_Position          35239 non-null  float64
 7   Plasticizing_Position     35239 non-null  float64
 8   Clamp_Open_Position       35239 non-null  float64
 9   Max_Injection_Speed       35239 non-null  float64
 10  Max_Screw_RPM             35239 non-null  float64
 11  Average_Screw_RPM         35239 non-null  float64
 12  Max_Injection_Pressure    35239 non-null  float64
 13  Max_Switch_Ove

,Unnamed: 0,Injection_Time,Filling_Time,Plasticizing_Time,Cycle_Time,Clamp_Close_Time,Cushion_Position,Plasticizing_Position,Clamp_Open_Position,Max_Injection_Speed,Max_Screw_RPM,Average_Screw_RPM,Max_Injection_Pressure,Max_Switch_Over_Pressure,Max_Back_Pressure,Average_Back_Pressure,Barrel_Temperature_1,Barrel_Temperature_2,Barrel_Temperature_3,Barrel_Temperature_4,Barrel_Temperature_5,Barrel_Temperature_6,Hopper_Temperature,Mold_Temperature_3,Mold_Temperature_4
0,114610,1.346829,-2.768862,0.961909,1.843294,1.029767,1.031111,0.849819,0.861854,2.489188,1.025136,1.031548,0.494886,0.345152,0.183351,0.221579,1.015093,0.919876,0.846380,0.637023,2.099671,1.025241,-0.551566,0.756435,0.600810
1,114688,1.360046,-2.720306,0.963280,1.847956,1.029767,1.031048,0.863559,0.861854,2.446877,1.025136,1.031548,0.497896,0.348230,0.201286,0.228419,1.020221,0.988615,0.874453,0.703109,2.099671,1.074495,-0.503460,0.818567,0.700620
2,114919,1.355641,-2.720306,0.966020,1.843294,1.029767,1.031080,0.864246,0.861854,2.446877,1.025136,1.031548,0.497896,0.351308,0.171395,0.221579,1.014360,0.862594,0.874453,0.637023,2.293882,0.975987,-0.599672,0.694303,0.574194
3,115304,1.360046,-2.720306,0.960539,1.843294,1.029767,1.031143,0.864246,0.861854,2.455339,1.025136,1.031548,0.497896,0.332839,0.183351,0.221579,1.017290,0.919876,0.944634,0.785717,2.099671,1.058077,-0.599672,0.531206,0.394536
4,115582,1.360046,-2.720306,0.967391,1.847956,1.029767,1.031048,0.864933,0.861854,2.446877,1.018644,1.031548,0.497896,0.326683,0.165417,0.221579,1.018756,0.851137,0.916562,0.670066,2.177355,0.992405,-0.599672,0.585571,0.434460


In [10]:
inspect_df(unlabeled_rg3, "Moldset Unlabeled RG3 Data")

📌 Moldset Unlabeled RG3 Data

[Shape]
(35941, 25)

[Info]
<class 'pandas.DataFrame'>
RangeIndex: 35941 entries, 0 to 35940
Data columns (total 25 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Unnamed: 0                35941 non-null  int64  
 1   Injection_Time            35941 non-null  float64
 2   Filling_Time              35941 non-null  float64
 3   Plasticizing_Time         35941 non-null  float64
 4   Cycle_Time                35941 non-null  float64
 5   Clamp_Close_Time          35941 non-null  float64
 6   Cushion_Position          35941 non-null  float64
 7   Plasticizing_Position     35941 non-null  float64
 8   Clamp_Open_Position       35941 non-null  float64
 9   Max_Injection_Speed       35941 non-null  float64
 10  Max_Screw_RPM             35941 non-null  float64
 11  Average_Screw_RPM         35941 non-null  float64
 12  Max_Injection_Pressure    35941 non-null  float64
 13  Max_Switch_Ove

,Unnamed: 0,Injection_Time,Filling_Time,Plasticizing_Time,Cycle_Time,Clamp_Close_Time,Cushion_Position,Plasticizing_Position,Clamp_Open_Position,Max_Injection_Speed,Max_Screw_RPM,Average_Screw_RPM,Max_Injection_Pressure,Max_Switch_Over_Pressure,Max_Back_Pressure,Average_Back_Pressure,Barrel_Temperature_1,Barrel_Temperature_2,Barrel_Temperature_3,Barrel_Temperature_4,Barrel_Temperature_5,Barrel_Temperature_6,Hopper_Temperature,Mold_Temperature_3,Mold_Temperature_4
0,119515,0.90919,1.662808,-1.131352,-1.174407,-1.262902,-1.314850,-1.026514,1.362091,-1.203070,-1.179145,-1.297009,-1.233889,-0.546310,-1.165335,-1.154268,-1.310547,-0.666957,-0.395393,0.088306,0.041839,0.945920,0.087427,-1.276966,-1.225083
1,119547,0.90919,1.662808,-1.131352,-1.174407,-1.262902,-1.314850,-1.026514,1.362091,-1.200666,-1.179145,-1.297009,-1.233889,-0.550067,-1.161183,-1.154268,-1.310547,-0.674680,-0.403577,0.078025,0.029691,0.925301,0.107476,-1.276966,-1.225083
2,119618,0.90919,1.662808,-1.147464,-1.174407,-1.262902,-1.315176,-1.026514,1.362091,-1.200666,-1.179145,-1.297009,-1.233889,-0.546310,-1.169488,-1.154268,-1.310547,-0.636064,-0.395393,0.078025,0.066135,0.904682,0.107476,-1.276966,-1.225083
3,119781,0.90919,1.662808,-1.131352,-1.174407,-1.262902,-1.314850,-1.026514,1.362091,-1.193454,-1.179145,-1.297009,-1.231013,-0.548189,-1.165335,-1.154268,-1.310547,-0.674680,-0.395393,0.067743,0.029691,0.863445,0.127524,-1.276966,-1.225083
4,120044,0.90919,1.662808,-1.131352,-1.174407,-1.262902,-1.314850,-1.026514,1.362091,-1.159797,-1.179145,-1.297009,-1.233889,-0.553825,-1.161183,-1.154268,-1.310547,-0.674680,-0.395393,0.088306,0.041839,0.945920,0.147573,-1.276966,-1.225083


### `supervised_label_cn7.csv`

In [11]:
inspect_df(supervised_cn7, "Supervised Label CN7 Data")

📌 Supervised Label CN7 Data

[Shape]
(6736, 26)

[Info]
<class 'pandas.DataFrame'>
RangeIndex: 6736 entries, 0 to 6735
Data columns (total 26 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Unnamed: 0                6736 non-null   int64  
 1   PassOrFail                6736 non-null   int64  
 2   Injection_Time            6736 non-null   float64
 3   Filling_Time              6736 non-null   float64
 4   Plasticizing_Time         6736 non-null   float64
 5   Cycle_Time                6736 non-null   float64
 6   Clamp_Close_Time          6736 non-null   float64
 7   Cushion_Position          6736 non-null   float64
 8   Plasticizing_Position     6736 non-null   float64
 9   Clamp_Open_Position       6736 non-null   float64
 10  Max_Injection_Speed       6736 non-null   float64
 11  Max_Screw_RPM             6736 non-null   float64
 12  Average_Screw_RPM         6736 non-null   float64
 13  Max_Injection_Pres

,Unnamed: 0,PassOrFail,Injection_Time,Filling_Time,Plasticizing_Time,Cycle_Time,Clamp_Close_Time,Cushion_Position,Plasticizing_Position,Clamp_Open_Position,Max_Injection_Speed,Max_Screw_RPM,Average_Screw_RPM,Max_Injection_Pressure,Max_Switch_Over_Pressure,Max_Back_Pressure,Average_Back_Pressure,Barrel_Temperature_1,Barrel_Temperature_2,Barrel_Temperature_3,Barrel_Temperature_4,Barrel_Temperature_5,Barrel_Temperature_6,Hopper_Temperature,Mold_Temperature_3,Mold_Temperature_4
0,0,0,9.60,4.48,16.91,59.580002,7.13,653.409973,68.839996,647.98999,55.299999,30.799999,292.500000,141.800003,136.899994,37.500000,59.299999,276.200012,275.500000,275.299988,270.799988,254.699997,229.500000,67.199997,24.799999,27.6
1,1,0,9.59,4.48,16.91,59.560001,7.13,653.419983,68.839996,647.98999,55.299999,31.000000,292.500000,141.800003,136.800003,37.500000,59.299999,276.500000,275.000000,275.399994,271.100006,254.899994,230.000000,66.900002,25.000000,27.6
2,2,0,9.58,4.46,16.90,59.580002,7.13,653.409973,68.839996,647.98999,55.599998,30.900000,292.500000,141.699997,136.399994,37.700001,59.400002,276.299988,275.299988,275.200012,271.399994,255.000000,230.000000,67.500000,25.000000,27.6
3,3,0,9.58,4.46,16.92,59.560001,7.13,653.409973,68.849998,647.98999,55.500000,30.600000,292.399994,141.800003,136.699997,37.400002,59.299999,275.799988,275.399994,275.000000,271.299988,255.000000,230.000000,67.000000,25.000000,27.6
4,4,0,9.57,4.45,16.91,59.520000,7.14,653.409973,68.830002,647.98999,55.700001,30.799999,292.500000,141.600006,136.399994,37.099998,59.099998,275.700012,274.899994,274.799988,270.799988,255.399994,230.100006,66.699997,24.799999,27.5


### `unlabeled_data.csv`

In [12]:
inspect_df(unlabeled, "Unlabeled Data")

📌 Unlabeled Data

[Shape]
(795315, 46)

[Info]
<class 'pandas.DataFrame'>
RangeIndex: 795315 entries, 0 to 795314
Data columns (total 46 columns):
 #   Column                    Non-Null Count   Dtype  
---  ------                    --------------   -----  
 0   Unnamed: 0                795315 non-null  int64  
 1   _id                       795315 non-null  str    
 2   TimeStamp                 795315 non-null  str    
 3   PART_FACT_PLAN_DATE       795315 non-null  str    
 4   PART_FACT_SERIAL          795315 non-null  int64  
 5   PART_NO                   795315 non-null  str    
 6   PART_NAME                 795315 non-null  str    
 7   EQUIP_CD                  795315 non-null  str    
 8   EQUIP_NAME                795315 non-null  str    
 9   ERR_FACT_QTY              795315 non-null  int64  
 10  Injection_Time            795315 non-null  float64
 11  Filling_Time              795315 non-null  float64
 12  Plasticizing_Time         795315 non-null  float64
 13  Cycle_Ti

,Unnamed: 0,_id,TimeStamp,PART_FACT_PLAN_DATE,PART_FACT_SERIAL,PART_NO,PART_NAME,EQUIP_CD,EQUIP_NAME,ERR_FACT_QTY,Injection_Time,Filling_Time,Plasticizing_Time,Cycle_Time,Clamp_Close_Time,Cushion_Position,Switch_Over_Position,Plasticizing_Position,Clamp_Open_Position,Max_Injection_Speed,Max_Screw_RPM,Average_Screw_RPM,Max_Injection_Pressure,Max_Switch_Over_Pressure,Max_Back_Pressure,Average_Back_Pressure,Barrel_Temperature_1,Barrel_Temperature_2,Barrel_Temperature_3,Barrel_Temperature_4,Barrel_Temperature_5,Barrel_Temperature_6,Barrel_Temperature_7,Hopper_Temperature,Mold_Temperature_1,Mold_Temperature_2,Mold_Temperature_3,Mold_Temperature_4,Mold_Temperature_5,Mold_Temperature_6,Mold_Temperature_7,Mold_Temperature_8,Mold_Temperature_9,Mold_Temperature_10,Mold_Temperature_11,Mold_Temperature_12
0,0,5f9136166e37704af878240e,2020-01-02 00:59:22,2020-01-02 오전 12:00:00,2,87285Q5000,"SP2 CVR ROOF RACK CTR, RH",S01,1800TON-우진,11,3.18,2.66,13.14,0.00,12.30,9.9,10.0,59.9,0.0,35.6,0.0,0.0,103.2,102.8,0.0,0.0,0.0,261.8,270.4,271.1,273.5,260.3,18.8,0.0,0,0,0.0,0.0,0,0,0,0,0,0,0,0
1,1,5f9136166e37704af878240f,2020-01-02 01:01:01,2020-01-02 오전 12:00:00,1,87275Q5000,"SP2 CVR ROOF RACK CTR, LH",S01,1800TON-우진,11,16.29,2.78,13.72,0.00,12.50,9.0,10.0,59.9,883.3,24.6,28.1,26.6,128.1,127.4,5.5,48.3,0.0,258.4,270.7,271.0,274.8,262.3,18.1,0.0,0,0,0.0,0.0,0,0,0,0,0,0,0,0
2,2,5f9136166e37704af8782410,2020-01-02 01:01:01,2020-01-02 오전 12:00:00,2,87285Q5000,"SP2 CVR ROOF RACK CTR, RH",S01,1800TON-우진,11,16.29,2.78,13.72,0.00,12.50,9.0,10.0,59.9,883.3,24.6,28.1,26.6,128.1,127.4,5.5,48.3,0.0,258.4,270.7,271.0,274.8,262.3,18.1,0.0,0,0,0.0,0.0,0,0,0,0,0,0,0,0
3,3,5f9136166e37704af8782411,2020-01-02 01:02:24,2020-01-02 오전 12:00:00,1,87275Q5000,"SP2 CVR ROOF RACK CTR, LH",S01,1800TON-우진,11,16.28,2.78,12.74,81.38,12.77,8.7,10.0,59.9,883.4,23.9,28.1,25.9,127.9,127.2,5.3,48.8,0.0,258.6,270.3,270.5,274.4,262.7,19.4,0.0,0,0,0.0,0.0,0,0,0,0,0,0,0,0
4,4,5f9136166e37704af8782412,2020-01-02 01:02:24,2020-01-02 오전 12:00:00,2,87285Q5000,"SP2 CVR ROOF RACK CTR, RH",S01,1800TON-우진,11,16.28,2.78,12.74,81.38,12.77,8.7,10.0,59.9,883.4,23.9,28.1,25.9,127.9,127.2,5.3,48.8,0.0,258.6,270.3,270.5,274.4,262.7,19.4,0.0,0,0,0.0,0.0,0,0,0,0,0,0,0,0


# 데이터 확인

### `unnamed: 0` 컬럼 제거
labeled, labeled_cn7, labeled_rg3, moldset_labeled, unlabeled_cn7, unlabeled_rg3, supervised_cn7, unlabeled

In [13]:
print(labeled_cn7.shape, labeled_rg3.shape, moldset_labeled.shape, unlabeled_cn7.shape, unlabeled_rg3.shape, supervised_cn7.shape, unlabeled.shape)

labeled_cn7.drop(columns=['Unnamed: 0'], inplace=True)
labeled_rg3.drop(columns=['Unnamed: 0'], inplace=True)
moldset_labeled.drop(columns=['Unnamed: 0'], inplace=True)
unlabeled_cn7.drop(columns=['Unnamed: 0'], inplace=True)
unlabeled_rg3.drop(columns=['Unnamed: 0'], inplace=True)
supervised_cn7.drop(columns=['Unnamed: 0'], inplace=True)
unlabeled.drop(columns=['Unnamed: 0'], inplace=True)

print(labeled_cn7.shape, labeled_rg3.shape, moldset_labeled.shape, unlabeled_cn7.shape, unlabeled_rg3.shape, supervised_cn7.shape, unlabeled.shape)

(1211, 26) (1182, 26) (2607, 47) (35239, 25) (35941, 25) (6736, 26) (795315, 46)
(1211, 25) (1182, 25) (2607, 46) (35239, 24) (35941, 24) (6736, 25) (795315, 45)


### `labeled_data.csv` -> `supervised_label_cn7.csv`

In [14]:
print("labeled_data.shape", labeled.shape)
print("supervised_label_cn7.shape", supervised_cn7.shape)

labeled_data.shape (7996, 45)
supervised_label_cn7.shape (6736, 25)


In [15]:
print("Columns:", set(labeled.columns) & set(supervised_cn7.columns))
print("Only in labeled_data:", set(labeled.columns) - set(supervised_cn7.columns))
print("Only in supervised_label_cn7:", set(supervised_cn7.columns) - set(labeled.columns))

Columns: {'Plasticizing_Position', 'Average_Screw_RPM', 'Mold_Temperature_3', 'Clamp_Open_Position', 'Barrel_Temperature_5', 'Cushion_Position', 'Max_Injection_Speed', 'Plasticizing_Time', 'Filling_Time', 'PassOrFail', 'Injection_Time', 'Max_Screw_RPM', 'Max_Back_Pressure', 'Barrel_Temperature_4', 'Barrel_Temperature_6', 'Barrel_Temperature_1', 'Max_Switch_Over_Pressure', 'Clamp_Close_Time', 'Max_Injection_Pressure', 'Barrel_Temperature_3', 'Cycle_Time', 'Hopper_Temperature', 'Average_Back_Pressure', 'Barrel_Temperature_2', 'Mold_Temperature_4'}
Only in labeled_data: {'PART_FACT_PLAN_DATE', 'Mold_Temperature_1', 'Mold_Temperature_10', 'Mold_Temperature_2', 'EQUIP_CD', 'Mold_Temperature_7', 'Mold_Temperature_11', 'Mold_Temperature_8', 'PART_NAME', 'Barrel_Temperature_7', 'Mold_Temperature_9', 'TimeStamp', 'EQUIP_NAME', 'PART_FACT_SERIAL', 'Mold_Temperature_6', 'Mold_Temperature_5', '_id', 'Mold_Temperature_12', 'Switch_Over_Position', 'Reason'}
Only in supervised_label_cn7: set()


> `supervised_labeled_cn7.csv`에만 있는 컬럼은 없음

#### `labeled_data`에서 제거된 컬럼 확인

In [ ]:
del_labeled = sorted(list(set(labeled.columns) - set(supervised_cn7.columns)))

for c in del_labeled:
    if c in ['_id', 'TimeStamp', 'PART_FACT_PLAN_DATE']:
        continue
    print(f"========== {c} ==========")
    print(labeled[c].value_counts())
    print("\n")

========== Barrel_Temperature_7 ==========
Barrel_Temperature_7
0.000000     7994
36.400002       2
Name: count, dtype: int64


========== EQUIP_CD ==========
EQUIP_CD
S14    7992
S01       2
S12       2
Name: count, dtype: int64


========== EQUIP_NAME ==========
EQUIP_NAME
650톤-우진2호기    7992
1800TON-우진       2
650톤-우진          2
Name: count, dtype: int64


========== Mold_Temperature_1 ==========
Mold_Temperature_1
0.0    7996
Name: count, dtype: int64


========== Mold_Temperature_10 ==========
Mold_Temperature_10
0.0    7996
Name: count, dtype: int64


========== Mold_Temperature_11 ==========
Mold_Temperature_11
0.0    7996
Name: count, dtype: int64


========== Mold_Temperature_12 ==========
Mold_Temperature_12
0.0    7996
Name: count, dtype: int64


========== Mold_Temperature_2 ==========
Mold_Temperature_2
0.0    7996
Name: count, dtype: int64


========== Mold_Temperature_5 ==========
Mold_Temperature_5
0.0    7996
Name: count, dtype: int64


========== Mold_Temperature_6 ===

- `_id` : 제조 공정 id **-> 불필요**
- `TimeStamp` : 시간 **-> 불필요**
- `PART_FACT_PLAN_DATE` : 생산을 지시한 날짜 **-> 불필요**
- `PART_FACT_SERIAL` : 생산을 지시한 품목에 대한 코드 부여 **-> 불필요**
- `PART_NAME` : 제품의 이름 **-> CN7 제품만 선택**
- `EQUIP_CD` : 생산한 사출기 호기 **-> S14만 선택**
- `EQUIP_NAME` : 생산한 사출기 모델명 **-> 650톤-우진2호기만 선택**
- `Reason` : 불량 유형 **-> 불필요, 결측치 많음**
- `Switch_Over_Position` : 고압(사출압)에서 보압으로 진행될 때의 위치 **-> 대부분 값이 0**
- `Barrel_Temperature_7` : 계량 및 사출 시 수지가 일정하게 용융(녹임)을 유지하기 위해 온도가 일정해야 한다 **-> 모든 값이 0**
- `Mold_Temperature_1~2, 5~12` : 재료 주입구의 온도 **-> 모든 값이 0**

In [17]:
labeled.groupby('PART_NAME')[['EQUIP_CD', 'EQUIP_NAME']].value_counts()

PART_NAME                  EQUIP_CD  EQUIP_NAME
CN7 W/S SIDE MLD'G LH      S14       650톤-우진2호기    3365
CN7 W/S SIDE MLD'G RH      S14       650톤-우진2호기    3371
JX1 W/S SIDE MLD'G RH      S12       650톤-우진          2
RG3 MOLD'G W/SHLD, LH      S14       650톤-우진2호기     628
RG3 MOLD'G W/SHLD, RH      S14       650톤-우진2호기     628
SP2 CVR ROOF RACK CTR, RH  S01       1800TON-우진       2
Name: count, dtype: int64

#### `labeled_data.csv` 전처리 후 `supervised_label_cn7.csv`와 비교

In [18]:
# CN7 제품만 선택
labeled_processed = labeled[labeled['PART_NAME'].isin(["CN7 W/S SIDE MLD'G LH", "CN7 W/S SIDE MLD'G RH"])].copy()

# 불필요한 컬럼 제거
labeled_processed.drop(columns=del_labeled, inplace=True)

# PassOrFail 인코딩
labeled_processed['PassOrFail'] = labeled_processed['PassOrFail'].map({'Y': 0, 'N': 1})
print(labeled_processed['PassOrFail'].unique())
print(labeled_processed['PassOrFail'].value_counts())

supervised_cn7.equals(labeled_processed)

[0 1]
PassOrFail
0    6697
1      39
Name: count, dtype: int64


False

In [19]:
print("Shape:", labeled_processed.shape == supervised_cn7.shape)
print("Columns:", labeled_processed.columns.equals(supervised_cn7.columns))
print("Index:", labeled_processed.index.equals(supervised_cn7.index))
print("Values:", labeled_processed.equals(supervised_cn7))

Shape: True
Columns: True
Index: False
Values: False


In [20]:
# labeled_data는 숫자형 컬럼의 소수점이 9자리까지 있고, supervised_label_cn7은 컬럼마다 다름

for decimals in range(10, 0, -1):

    df1 = labeled_processed.copy()
    df2 = supervised_cn7.copy()

    num_cols = df1.select_dtypes(include="number").columns

    df1[num_cols] = df1[num_cols].round(decimals)
    df2[num_cols] = df2[num_cols].round(decimals)

    vc1 = df1.value_counts(dropna=False).sort_index()
    vc2 = df2.value_counts(dropna=False).sort_index()

    print(
        f"round({decimals}):",
        vc1.equals(vc2)
    )

round(10): False
round(9): False
round(8): False
round(7): False
round(6): True
round(5): True
round(4): True
round(3): True
round(2): True
round(1): True


In [21]:
# 소수점 6자리까지 반올림해서 다시 비교
df1 = labeled_processed.copy()
df2 = supervised_cn7.copy()

num_cols = df1.select_dtypes(include="number").columns

df1[num_cols] = df1[num_cols].round(6)
df2[num_cols] = df2[num_cols].round(6)

vc1 = df1.value_counts(dropna=False).sort_index()
vc2 = df2.value_counts(dropna=False).sort_index()

print(vc1.equals(vc2))

True


> `labeled_data.csv` -> `supervised_label_cn7.csv`가 맞음

#### supervised_label_rg3 ?

In [22]:
# RG3 제품만 선택
supervised_rg3 = labeled[labeled['PART_NAME'].isin(["RG3 MOLD'G W/SHLD, LH", "RG3 MOLD'G W/SHLD, RH"])].copy()

# 불필요한 컬럼 제거
supervised_rg3.drop(columns=del_labeled, inplace=True)

# PassOrFail 인코딩
supervised_rg3['PassOrFail'] = supervised_rg3['PassOrFail'].map({'Y': 0, 'N': 1})
print(supervised_rg3['PassOrFail'].unique())

# 확인
print(supervised_rg3.shape)
supervised_rg3.info()

[0 1]
(1256, 25)
<class 'pandas.DataFrame'>
Index: 1256 entries, 1211 to 5231
Data columns (total 25 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   PassOrFail                1256 non-null   int64  
 1   Injection_Time            1256 non-null   float64
 2   Filling_Time              1256 non-null   float64
 3   Plasticizing_Time         1256 non-null   float64
 4   Cycle_Time                1256 non-null   float64
 5   Clamp_Close_Time          1256 non-null   float64
 6   Cushion_Position          1256 non-null   float64
 7   Plasticizing_Position     1256 non-null   float64
 8   Clamp_Open_Position       1256 non-null   float64
 9   Max_Injection_Speed       1256 non-null   float64
 10  Max_Screw_RPM             1256 non-null   float64
 11  Average_Screw_RPM         1256 non-null   float64
 12  Max_Injection_Pressure    1256 non-null   float64
 13  Max_Switch_Over_Pressure  1256 non-null   float64
 14  Max_

In [41]:
print(labeled_processed.shape)
print("CN7 양품 수:", labeled_processed[labeled_processed['PassOrFail']==0].shape[0])
print("CN7 불량품 수:", labeled_processed[labeled_processed['PassOrFail']==1].shape[0])
print("="*20)
print(supervised_rg3.shape)
print("RG3 양품 수:", supervised_rg3[supervised_rg3['PassOrFail']==0].shape[0])
print("RG3 불량품 수:", supervised_rg3[supervised_rg3['PassOrFail']==1].shape[0])

(6736, 25)
CN7 양품 수: 6697
CN7 불량품 수: 39
(1256, 25)
RG3 양품 수: 1224
RG3 불량품 수: 32


### `moldset_labeled.csv` -> `moldset_labeled_cn7.csv`, `moldset_labeled_rg3.csv`


In [23]:
print("moldset_labeled.shape", moldset_labeled.shape)
print("moldset_labeled_cn7.shape", labeled_cn7.shape)
print("moldset_labeled_rg3.shape", labeled_rg3.shape)

moldset_labeled.shape (2607, 46)
moldset_labeled_cn7.shape (1211, 25)
moldset_labeled_rg3.shape (1182, 25)


In [24]:
set(labeled_cn7.columns) == set(labeled_rg3.columns)

True

In [25]:
print("Only in moldset_labeled:", set(moldset_labeled.columns) - set(labeled_cn7.columns))
print("Only in moldset_labeled_cn7/rg3:", set(labeled_cn7.columns) - set(moldset_labeled.columns))

Only in moldset_labeled: {'PART_FACT_PLAN_DATE', 'PART_NO', 'Mold_Temperature_1', 'Mold_Temperature_10', 'Mold_Temperature_2', 'EQUIP_CD', 'Mold_Temperature_7', 'Mold_Temperature_11', 'Mold_Temperature_8', 'PART_NAME', 'Barrel_Temperature_7', 'Mold_Temperature_9', 'TimeStamp', 'EQUIP_NAME', 'PART_FACT_SERIAL', 'Mold_Temperature_6', 'Mold_Temperature_5', '_id', 'Mold_Temperature_12', 'Switch_Over_Position', 'Reason'}
Only in moldset_labeled_cn7/rg3: set()


#### `moldset_labeled`에서 제거된 컬럼 확인

In [26]:
del_moldset = sorted(list(set(moldset_labeled.columns) - set(supervised_cn7.columns)))

for c in del_moldset:
    if c in ['_id', 'TimeStamp', 'PART_FACT_PLAN_DATE']:
        continue
    print(f"========== {c} ==========")
    print(moldset_labeled[c].value_counts())
    print("\n")

========== Barrel_Temperature_7 ==========
Barrel_Temperature_7
0.0    2607
Name: count, dtype: int64


========== EQUIP_CD ==========
EQUIP_CD
S14    2607
Name: count, dtype: int64


========== EQUIP_NAME ==========
EQUIP_NAME
650톤-우진2호기    2607
Name: count, dtype: int64


========== Mold_Temperature_1 ==========
Mold_Temperature_1
0.0    2607
Name: count, dtype: int64


========== Mold_Temperature_10 ==========
Mold_Temperature_10
0.0    2607
Name: count, dtype: int64


========== Mold_Temperature_11 ==========
Mold_Temperature_11
0.0    2607
Name: count, dtype: int64


========== Mold_Temperature_12 ==========
Mold_Temperature_12
0.0    2607
Name: count, dtype: int64


========== Mold_Temperature_2 ==========
Mold_Temperature_2
0.0    2607
Name: count, dtype: int64


========== Mold_Temperature_5 ==========
Mold_Temperature_5
0.0    2607
Name: count, dtype: int64


========== Mold_Temperature_6 ==========
Mold_Temperature_6
0.0    2607
Name: count, dtype: int64


========== Mold_Tem

- `TimeStamp` : 시간 **-> 불필요**
- `PART_FACT_PLAN_DATE` : 생산을 지시한 날짜 **-> 불필요**
- `PART_FACT_SERIAL` : 생산을 지시한 품목에 대한 코드 부여 **-> 불필요**
- `PART_NAME` : 제품의 이름 **-> CN7/RG3 제품만 존재**
- `PART_NO` : 제품의 모델 코드 **-> 불필요, PART_NAME과 같은 분류**
- `EQUIP_CD` : 생산한 사출기 호기 **-> S14만 존재**
- `EQUIP_NAME` : 생산한 사출기 모델명 **-> 650톤-우진2호기만 존재**
- `Reason` : 불량 유형 **-> 불필요, 결측치 많음**
- `Switch_Over_Position` : 고압(사출압)에서 보압으로 진행될 때의 위치 **-> 대부분 값이 0**
- `Barrel_Temperature_7` : 계량 및 사출 시 수지가 일정하게 용융(녹임)을 유지하기 위해 온도가 일정해야 한다 **-> 모든 값이 0**
- `Mold_Temperature_1~2, 5~12` : 재료 주입구의 온도 **-> 모든 값이 0**

In [27]:
moldset_labeled['PART_NO'].value_counts()

PART_NO
86141AA000    713
86131AA000    712
86131T1000    591
86141T1000    591
Name: count, dtype: int64

#### `moldset_labeled.csv` 전처리 후 `moldset_labeled_cn7/rg3.csv`와 비교

In [28]:
moldset_labeled['PassOrFail'].unique()

array([0, 1])

In [29]:
# 제품 선택
moldset_processed_cn7 = moldset_labeled[moldset_labeled['PART_NAME'].isin(["CN7 W/S SIDE MLD'G LH", "CN7 W/S SIDE MLD'G RH"])].copy()
moldset_processed_rg3 = moldset_labeled[moldset_labeled['PART_NAME'].isin(["RG3 MOLD'G W/SHLD, LH", "RG3 MOLD'G W/SHLD, RH"])].copy()

# 불필요한 컬럼 제거
moldset_processed_cn7.drop(columns=del_moldset, inplace=True)
moldset_processed_rg3.drop(columns=del_moldset, inplace=True)

# PassOrFail 인코딩 불필요

print("moldset_processed_cn7.shape", moldset_processed_cn7.shape)
print("moldset_processed_rg3.shape", moldset_processed_rg3.shape)

moldset_processed_cn7.shape (1425, 25)
moldset_processed_rg3.shape (1182, 25)


> `moldset_labeled_cn7`은 (1211, 25)인데 `moldset_labeled_cn7.shape`는 (1425, 25)

In [30]:
print("CN7 양품 수:", moldset_processed_cn7[moldset_processed_cn7['PassOrFail']==0].shape[0])
print("CN7 불량품 수:", moldset_processed_cn7[moldset_processed_cn7['PassOrFail']==1].shape[0])
print("="*20)
print("RG3 양품 수:", moldset_processed_rg3[moldset_processed_rg3['PassOrFail']==0].shape[0])
print("RG3 불량품 수:", moldset_processed_rg3[moldset_processed_rg3['PassOrFail']==1].shape[0])

CN7 양품 수: 1398
CN7 불량품 수: 27
RG3 양품 수: 1157
RG3 불량품 수: 25


In [31]:
from sklearn.preprocessing import StandardScaler
for column in moldset_processed_rg3.columns:
    if column != 'PassOrFail':
        sc = StandardScaler()
        moldset_processed_rg3[column] = sc.fit_transform(moldset_processed_rg3[[column]])

In [ ]:
# moldset_labeled_rg3.csv만 비교해보기

for decimals in range(15, 0, -1):

    df1 = labeled_rg3.copy()
    df2 = moldset_processed_rg3.copy()

    num_cols = df1.select_dtypes(include="number").columns

    df1[num_cols] = df1[num_cols].round(decimals)
    df2[num_cols] = df2[num_cols].round(decimals)

    vc1 = df1.value_counts(dropna=False).sort_index()
    vc2 = df2.value_counts(dropna=False).sort_index()

    print(
        f"round({decimals}):",
        vc1.equals(vc2)
    )

round(15): False
round(14): False
round(13): False
round(12): False
round(11): True
round(10): True
round(9): True
round(8): True
round(7): True
round(6): True
round(5): True
round(4): True
round(3): True
round(2): True
round(1): True
